### PR TITLE
Parameterize connectivity tests

### PR DESCRIPTION
**Pull request by birm**
re #1184

This attempts to organize/parameterize tests in connectivty.jl

### CODE REVIEW

File: test/community/label_propagation.jl [hunk 1]
```diff
@@ -4,6 +4,7 @@
     for g in testgraphs(g10)
         z = copy(g)
         for k = 2:5
+            Random.seed!(0)
```
Comment (sbromberger) [inline test/community/label_propagation.jl]: revert this. It's a known issue but forcing a seed in an unrelated piece of code is not good for this PR.

**Comment by birm** [comment]
While not strictily related, it seems as though the label propogation test sometimes fails (see the build for `c91f705` or below); I've set the random seed which should fix that.

Example:
```julia
julia> n = 10
10

julia> g10 = complete_graph(n)
{10, 45} undirected simple Int64 graph

julia> for zzz in collect(1:100)
         for g in testgraphs(g10)
             z = copy(g)
             for k = 2:5
                 z = blockdiag(z, g)
                 add_edge!(z, (k - 1) * n, k * n)
                 c, ch = label_propagation(z)
                 a = collect(n:n:(k * n))
                 a = Int[div(i - 1, n) + 1 for i = 1:(k * n)]
                 if ! (a==c)
                   show(stdout, "text/plain", k)
                   show(stdout, "text/plain", a)
                   show(stdout, "text/plain", c)
                   show(stdout, "text/plain", a==c)
                 end
             end
         end
       end
440-element Array{Int64,1}:
 1
 1
 1
 1
 1
 1
 1
 1
 1
 1
 2
 2
 2
 2
 2
 2
 2
 2
 2
 2
 3
 3
 3
 3
 3
 3
 3
 3
 3
 3
 4
 4
 4
 4
 4
 4
 4
 4
 4
 440-element Array{UInt8,1}:
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x01
 0x02
 0x02
 0x02
 0x02
 0x02
 0x02
 0x02
 0x02
 0x02
 0x02
 0x03
 0x03
 0x03
 0x03
 0x03
 0x03
 0x03
 0x03
 0x03
 0x03
 false

```

File: test/community/label_propagation.jl (hunk 1 above)
Comment (birm) [reply test/community/label_propagation.jl]: Ok; apologies for going off topic.

**Comment by sbromberger** [comment]
> While not strictily related, it seems as though the label propogation test sometimes fails

It does, but including the fix in an unrelated (and already approved) PR is not the best way to deal with this. If you have failures, getting a contributor to restart Travis is the best way to move forward at this point.

File: test/community/label_propagation.jl (hunk 1 above)
Comment (sbromberger) [reply test/community/label_propagation.jl]: No worries. If you'd like to make a PR that fixes this, please do so; we'd want to loop in @jpfairbanks and others on the review to see whether this approach (fixed seed) makes sense or whether there's a better way to handle failures in random functions.

**Comment by birm** [comment]
@sbromberger Would you welcome a separate PR with the same or similar fix?

**Comment by sbromberger** [comment]
> Would you welcome a separate PR with the same or similar fix?

Absolutely, we'd want to loop in some other people for a discussion as to whether there's a better way to test functions whose output may fail randomly.

**Pull request merged**